doc/feature/transferFunctions.md

```diff
@@ -0,0 +1,238 @@
+
+Control points are always in the absolute scale, that means that the x coordinate
+refers to the scale of the source variable and the y coordinate refers to the
+scale of the target attribute. The range of RGBA channels will be [0, 1] unless
+an application specifies something different (e.g. in physically based TFs)
+For example, a simple black to white colormap
+for neuron membrane voltage
+could be described by the control points (-80, 0), (-10, 1)
+
+Transfer functions or colormaps cannot be partially updated, the messages
+always describe the entire function. In cases where the graphical editor allows
+remapping a color map to a different input range the editor is responsible
+of finding the adjusted x positions of the control points. For convenience, the
+editor may store colormaps using normalized coordinates for the x axis and then
+recalculate the positions when applied to an application.
+
+### Messages
+
+    namespace lexis.render.detail; // subnamespace needed to avoid name clashes
+
+    table ControlPoint
+    {
+        x: float;
+        y: float;
+    }
+
+    table TransferFunction
+    {
+        attribute: string;
+        points: [ControlPoint];
+    }
+
+    table ColorMap
+    {
+        red: [ControlPoint];
+        green: [ControlPoint];
+        blue: [ControlPoint];
+        alpha: [ControlPoint];
+    }
+
+### C++ helper classes
+
+    /**
+     *  Wraps a lexis.render.detail.ControlMap
+     */
+    class ColorMap
+    {
+        enum class Channel
+        {
+            red       = 0,
+            green     = 1,
+            blue      = 2,
+            alpha     = 3,
+        };
+
+        /**
+         * Constructs a color map from a file ( *.lba, *.lbb )
+         * @param filename name of the color map file
+         */
+        ColorMap( const std::string& filename );
+
+        /**
+         * Constructs a default color map with a given range.
+         *
+         * The default color map is a list of 50 points from the Viridis color
+         * map from Matplotlib (http://bids.github.io/colormap/).
+         */
+        ColorMap( float min, float max );
+
+       /**
+        * Compute a linear sampling of the color map.
+        *
+        * @param data A length * 4 array of T that will be filled with the color
+        *        values in RGBARGBA.... form. If there are no control points
+        *        for a channel, the returned colors will have 0 for that
+        *        channel. If there is only one control point, all samples will
+        *        have the same color for that channel.  The ith sample will
+        *        contain the color for min/(max-min) * i/(length-1) where min
+        *        and max are the minimum and maximum x values of all control
+        *        points.
+        *        Valid types for T are all numerical types.
+        * @param length number of sample points
+        */
+        template< typename T > sampleColors( T* data, size_t length ) const;
```

Comment:
void

This function fills a plain array because it doesn't want to make any assumption about the storage to be used. The target memory could very well be something you can't store in a unique_ptr, not to say a std::vector.